src/classes/STG_PanelOppNaming_CTRL.cls

```diff
@@ -183,21 +206,92 @@ public with sharing class STG_PanelOppNaming_CTRL extends STG_Panel {
     }
 
     /*********************************************************************************************************
-    * @description The list of SelectOptions for Opportunity Attribution 
+    * @description The list of SelectOptions for Opportunity Attribution
     */
-    static public list<SelectOption> listSOOpportunityAttribution {
-        get {    
+    public static list<SelectOption> listSOOpportunityAttribution {
+        get {
             if (listSOOpportunityAttribution == null) {
-                listSOOpportunityAttribution = new list<SelectOption>();
-                listSOOpportunityAttribution.add(new SelectOption(Label.oppNamingBoth, Label.oppNamingBoth));
-                listSOOpportunityAttribution.add(new SelectOption(Label.oppNamingIndividual, Label.oppNamingIndividual));
-                listSOOpportunityAttribution.add(new SelectOption(Label.oppNamingOrganizational, Label.oppNamingOrganizational));
+                listSOOpportunityAttribution = new AttributionSettings().getSelectOptions();
             }
             return listSOOpportunityAttribution;
         }
         private set;
     }
 
+
+    /***
+    * @description Handles Opportunity Naming Settings Attribution options
+    */
+    public class AttributionSettings {
```

Comment:
New version of the class that needs to be tested outside of the other existing/new methods:
```
/***
    * @description Handles Opportunity Naming Settings Attribution options
    */
    public class AttributionSettings {

        public String selectedOption { get; private set; }
        public String selectedLabel { get; private set; }

        /***
        * @description Contains the Opp Naming Settings Attribution API name to the custom label mapping
        */
        private Map<String, String> labelByOption {
            get {
                if (labelByOption == null) {
                    labelByOption = new Map<String, String>{
                        AttributionOptions.Both.name() => System.Label.OppNamingBoth,
                        AttributionOptions.Contact_Donations.name() => System.Label.oppNamingIndividual,
                        AttributionOptions.Organization_Donations.name() => System.Label.oppNamingOrganizational
                    };
                }
                return labelByOption;
            }
            set;
        }

        /***
        * @description Contains the Opp Naming Settings Attribution custom label to the API name mapping
        */
        private Map<String, String> optionByLabel {
            get {
                if (optionByLabel == null) {
                    optionByLabel = new Map<String, String>{
                        System.Label.OppNamingBoth => AttributionOptions.Both.name(),
                        'Both' => AttributionOptions.Both.name(),

                        System.Label.oppNamingIndividual => AttributionOptions.Contact_Donations.name(),
                        'Contact Donations' => AttributionOptions.Contact_Donations.name(),

                        System.Label.oppNamingOrganizational => AttributionOptions.Organization_Donations.name(),
                        'Organization Donations' => AttributionOptions.Organization_Donations.name()
                    };
                }
                return optionByLabel;
            }
            set;
        }

        /***
        * @description Sets attribution value as configured in the Opportunity Naming Settings
        */
        public AttributionSettings withSelected(String option) {
            this.selectedOption = optionByLabel.containsKey(option) ? optionByLabel.get(option) : option;
            this.selectedLabel = getLabel(this.selectedOption);
            return this;
        }

        /***
        * @description Generate the select options for attribution settings
        */
        public List<SelectOption> getSelectOptions() {
            List<SelectOption> options = new List<SelectOption>();

            for (AttributionOptions option : AttributionOptions.values()) {
                options.add(new SelectOption(option.name(), getLabel(option.name())));
            }

            return options;
        }

        /***
        * @description Map the attribution option to the custom label used in Opp naming or settings field option display
        */
        public String getLabel(String option) {
            String label = labelByOption.get(option);

            return String.isBlank(label) ? '' : label;
        }

        /***
        * @description Returns true if attribution is applicable to both Contact and Organization Donations
        */
        public Boolean isBoth() {
            return selectedOption == AttributionOptions.Both.name();
        }

    }
```